dapp/src/insurance/testflightdelay.sol

```diff
@@ -0,0 +1,745 @@
+	enum oraclizeState { ForUnderwriting, ForPayout }
+
+	event LOG_PolicyApplied(
+		uint policyId,
+		address customer,
+		string carrierFlightNumber,
+		uint premium
+	);
+	event LOG_PolicyAccepted(
+		uint policyId,
+		uint statistics0,
+		uint statistics1,
+		uint statistics2,
+		uint statistics3,
+		uint statistics4,
+		uint statistics5
+	);
+	event LOG_PolicyRevoked(
+		uint policyId
+	);
+	event LOG_PolicyPaidOut(
+		uint policyId,
+		uint amount
+	);
+	event LOG_PolicyExpired(
+		uint policyId
+	);
+	event LOG_PolicyDeclined(
+		uint policyId,
+		bytes32 reason
+	);
+	event LOG_PolicyManualPayout(
+		uint policyId,
+		bytes32 reason
+	);
+	event LOG_SendFail(
+		uint policyId,
+		bytes32 reason
+	);
+	event LOG_OraclizeCall(
+		uint policyId,
+		bytes32 queryId,
+		string oraclize_url
+	);
+	event LOG_OraclizeCallback(
+		uint policyId,
+		bytes32 queryId,
+		string result,
+		bytes proof
+	);
+	event LOG_HealthCheck(
+		bytes32 message, 
+		int diff,
+		uint balance,
+		int ledgerBalance 
+	);
+
+	// some general constants for the system:
+	// minimum observations for valid prediction
+	uint constant minObservations 			= 10;
+	// minimum premium to cover costs
+	uint constant minPremium 				= 500 finney;
+	// maximum premium
+	uint constant maxPremium 				= 5 ether;
+	// maximum payout
+	uint constant maxPayout 				= 200 ether;
+	// maximum number of identical risks
+	uint8 constant maxIdenticalRisks		= 10;
+	// 1 percent for DAO, 1 percent for maintainer
+	uint8 constant rewardPercent 			= 2;
+	// reserve for tail risks
+	uint8 constant reservePercent 			= 1;
+	// the weight pattern; in future versions this may become part of the policy struct.
+	// currently can't be constant because of compiler restrictions
+	// weightPattern[0] is not used, just to be consistent
+    uint8[6] weightPattern 					= [0, 10,20,30,50,50];
+	// Deadline for acceptance of policies: Mon, 26 Sep 2016 12:00:00 GMT
+	uint contractDeadline 					= 1474891200; 
+
+	// account numbers for the internal ledger:
+	// sum of all Premiums of all currently active policies
+	uint8 constant acc_Premium 				= 0;
+	// Risk fund; serves as reserve for tail risks
+	uint8 constant acc_RiskFund 			= 1;
+	// sum of all payed out policies
+	uint8 constant acc_Payout 				= 2;
+	// the balance of the contract (negative!!)
+	uint8 constant acc_Balance 				= 3;
+	// the reward account for DAO and maintainer
+	uint8 constant acc_Reward 				= 4;
+	// oraclize costs
+	uint8 constant acc_OraclizeCosts 		= 5;
+	// when adding more accounts, remember to increase ledger array length
+
+	// Maintenance modes 
+	uint8 constant maintenance_None      	= 0;
+	uint8 constant maintenance_BalTooHigh 	= 1;
+	uint8 constant maintenance_Emergency 	= 255;
+	
+	
+	// gas Constants for oraclize
+	uint constant oraclizeGas 				= 500000;
+
+	// URLs and query strings for oraclize
+
+	string constant oraclize_RatingsBaseUrl =
+		"[URL] json(https://api.flightstats.com/flex/ratings/rest/v1/json/flight/";
```

Comment:
The whole contract makes use of oraclize to perform external calls, I am sure there is a way to do everything on chain and in a simpler way